### PR TITLE
AXLE-48 Store investor sessions per acount + add hub signout method

### PR DESCRIPTION
**Pull request by iliyan-georgiev-ph**

### CODE REVIEW

File: src/Axle/Services/SessionLifecycleService.cs [hunk 1]
```diff
@@ -119,6 +119,8 @@ public void CloseConnection(string connectionId)
 
         public async Task<TerminateSessionResponse> TerminateSession(
             string userName,
```
Comment (Waqas-Zamir) [inline src/Axle/Services/SessionLifecycleService.cs]: session termination is only possible by account id

File: src/Axle/Hubs/SessionHub.cs [hunk 2]
```diff
@@ -65,6 +65,17 @@ public override Task OnDisconnectedAsync(Exception exception)
             return base.OnDisconnectedAsync(exception);
         }
 
+        public async Task<bool> SignOut()
+        {
+            var userName = this.Context.User.FindFirst(JwtClaimTypes.Name).Value;
+            var accountId = this.Context.GetHttpContext().Request.Query["account_id"];
+            var isSupportUser = this.Context.User.IsSupportUser(accountId);
+
+            var response = await this.sessionLifecycleService.TerminateSession(userName, accountId, isSupportUser, SessionActivityType.SignOut);
```
Comment (Waqas-Zamir) [inline src/Axle/Hubs/SessionHub.cs]: using TerminateSession for signout violates SRP, better to have two separate methods

File: src/Axle/Services/SessionLifecycleService.cs [hunk 3]
```diff
@@ -127,22 +129,27 @@ public void CloseConnection(string connectionId)
...
+                        ErrorMessage = $"No session found for the user: [{userName}] and account: [{accountId}]"
                     };
                 }
 
-                this.logger.LogInformation($"Terminating session: [{userInfo.SessionId}] for user: [{userName}]");
+                this.logger.LogInformation($"Terminating session: [{userInfo.SessionId}] for user: [{userName}] and account: [{accountId}]");
 
                 await this.TerminateSession(userInfo, reason);
 
                 this.logger.LogWarning(StatusCode.WN_ATH_701.ToMessage());
```
Comment (Waqas-Zamir) [inline src/Axle/Services/SessionLifecycleService.cs]: Logging terminate session warning isn't correct for signout

File: src/Axle/Services/SessionLifecycleService.cs [hunk 4]
```diff
@@ -168,7 +175,7 @@ public void CloseConnection(string connectionId)
 
         public async Task TerminateSession(Session userInfo, SessionActivityType reason)
         {
-            this.sessionRepository.Remove(userInfo.SessionId, userInfo.UserName);
+            this.sessionRepository.Remove(userInfo.SessionId, userInfo.UserName, userInfo.AccountId);
             await this.tokenRevocationService.RevokeAccessToken(userInfo.AccessToken, userInfo.ClientId);
```
Comment (Waqas-Zamir) [inline src/Axle/Services/SessionLifecycleService.cs]: do we need to revoke access token for signout?

File: src/Axle/Hubs/SessionHub.cs (hunk 2 above)
Comment (iliyan-georgiev-ph) [reply src/Axle/Hubs/SessionHub.cs]: TerminateSession's purpose is not only for manual termination, but for all needs to end a session, as indicated by the reason parameter - it's used for session timeout, different device login, manual termination and now sign out

File: src/Axle/Services/SessionLifecycleService.cs (hunk 1 above)
Comment (iliyan-georgiev-ph) [reply src/Axle/Services/SessionLifecycleService.cs]: Support users also have sessions, hence the possibility to end the session using only the user ID

File: src/Axle/Services/SessionLifecycleService.cs (hunk 4 above)
Comment (iliyan-georgiev-ph) [reply src/Axle/Services/SessionLifecycleService.cs]: Revoking the access token is part of the sign out process I believe

Comment (Waqas-Zamir) [reply src/Axle/Services/SessionLifecycleService.cs]: calling signout from FE and revoking access token from BE approximately simultaneously will result in unexpected redirect after signout.

1. Signout from FE is called first then user will see proper logout page
2. If access token is revoked before signout from FE then user may see some other page from bouncer

May be we should revoke access token other than signout reason

**Pull request merged**